src/ansible_compat/runtime.py

```diff
@@ -284,23 +284,38 @@ def install_requirements(self, requirement: str, retry: bool = False) -> None:
                 raise AnsibleCommandError(run)
 
     def prepare_environment(
-        self, required_collections: Optional[Dict[str, str]] = None, retry: bool = False
+        self,
+        required_collections: Optional[Dict[str, str]] = None,
+        retry: bool = False,
+        install_local: bool = False,
     ) -> None:
         """Make dependencies available if needed."""
         if required_collections is None:
             required_collections = {}
 
         self.install_requirements("requirements.yml", retry=retry)
 
+        destination = f"{self.cache_dir}/collections" if self.cache_dir else None
         for name, min_version in required_collections.items():
             self.install_collection(
                 f"{name}:>={min_version}",
-                destination=f"{self.cache_dir}/collections" if self.cache_dir else None,
+                destination=destination,
             )
 
         self._prepare_ansible_paths()
-        # install role if current project looks like a standalone role
-        self._install_galaxy_role(self.project_dir, ignore_errors=True)
+
+        if not install_local:
+            return
+
+        if os.path.exists("galaxy.yml"):
+            # molecule scenario within a collection
```

Comment:
This is not 100% correct since there is no rule that the `molecule` directory must live at the root of the collection.

Reply:
We only ever supported at repository root and inside a role. I am keep to add support to add heuristics for finding any galaxy.yml at upper levels. It would add extra complexity and encourage people to invent new layouts, which will likely translate to new bugs.

Do you have a specific case that worked before and no longer works after we add this change?

Reply:
I was just commenting on the comment here, not the functionality (the functionality is fine with me). But rereading it now, it does not bother me that much anymore, so I guess we can keep it as is.